plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2996,6 +2997,12 @@ bool producer_plugin_impl::push_read_only_transaction(transaction_metadata_ptr t
       if( !_ro_in_read_only_mode ) {
          chain.set_db_read_only_mode();
       }
+
+      if ( app().executor().is_write_window() ) {
```

Comment:
Seems like this should be `!_ro_in_read_only_mode` to be consistent. Same for below.

Reply:
really it looks like `_ro_in_read_only_mode` and `app().executor().is_read_window()` are tracking the same state. Why the duplication?

Reply:
We cannot use `!_ro_in_read_only_mode` to tell if a read-only trx is running exclusively on the main thread (in write window) or running on the main thread in parallel with read-only threads (in read-window).
For any read-only trxs in either read window or write window,
`_ro_in_read_only_mode` is always true. So `!_ro_in_read_only_mode` is always false.

Reply:
Maybe I'm having a slow start this Monday, but it appears to me that `!_ro_in_read_only_mode` <==> `app().executor().is_write_window()`. As such, I think we should remove `_ro_in_read_only_mode` and only use `app().executor().is_write_window()`

`_ro_in_read_only_mode` is set at the same time/place as `app().executor().set_to_*_window()`.

Reply:
I was overlooking and thought `_ro_in_read_only_mode` as chain's `db_read_only_mode`. Will replace `_ro_in_read_only_mode` with `app().executor().is_write_window()`.